File: Sorting/SelectionSort.cpp
// Selection Sort ALgorithm

#include<iostream>
using namespace std;


void SelectionSort(int Arr[],int iSize)
{

    int i = 0,j = 0,temp,min;
     
    for(i=0;i<iSize-1;i++)
    {
        min = i;
        for(j=i+1;j<iSize;j++)
        {
            if(Arr[j] < Arr[min])
            {
                min = j;
            }
        }
        if(min !=i)
        {
        temp = Arr[i];
        Arr[i] = Arr[min];
        Arr[min] = temp;
        }
    }

    cout<<"Sorted Array in Ascending Order : ";
    for(i=0;i<iSize;i++)
    {
        cout<<Arr[i]<<"\t";
    }

}
int main()
{
    int iSize = 0;
    int Arr[iSize];

    cout<<"Enter the Size of Array :\n";
    cin>>iSize;

    cout<<"Enter the Element of an Array \n";
    for(int i=0;i<iSize;i++)
    {
        cin>>Arr[i];
    }

    SelectionSort(Arr,iSize);
}

File: Sorting/SelectionSort.java

/*
 *   In Selection Sort is Simple Comparison based sorting algorithm that 
 *  divides input into two parts sorted and unsorted sorted at left and unsorted at right
 *   
 *  we find minimum element in left part and swap it with current element
 */

import java.util.Scanner;


class Sort 
{
    void  SelSort(int Arr[])
    { 

        int min = 0 ,temp = 0;
         for(int i=0;i<Arr.length-1;i++)
         { 
            min = i;
            for(int j=i+1;j<Arr.length;j++)
            {
              if(Arr[j] < Arr[min])
              {
                min = j;
              }
            }
            temp = Arr[i];
            Arr[i] = Arr[min];
            Arr[min] = temp;
         }


    }
}
class SelectionSort
{
    public static void main(String args[])
    {
          int n = 0;
          
          System.out.println("Enter the Size of An Array : ");
          Scanner sobj = new Scanner(System.in);

          n = sobj.nextInt();
           
          int Arr[] = new int[n];
          for(int i=0;i<n;i++)
          {
            Arr[i] = sobj.nextInt();
          }

    }
}